test/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/representee/RepresenteeControllerSpec.scala

```diff
@@ -416,6 +418,284 @@ class RepresenteeControllerSpec
+              doc
+                .select("#content > article > form")
+                .attr("action") shouldBe routes.RepresenteeController.enterDateOfDeathSubmit().url
+              doc.select("#returnToSummaryLink").text() shouldBe (
+                if (expectReturnToSummaryLink) messageFromMessageKey("returns.return-to-summary-link")
+                else ""
+              )
+              expectedPrepopulatedValue.foreach { date =>
+                doc.select("#dateOfDeath-day").attr("value")   shouldBe date.value.getDayOfMonth.toString
+                doc.select("#dateOfDeath-month").attr("value") shouldBe date.value.getMonthValue.toString
+                doc.select("#dateOfDeath-year").attr("value")  shouldBe date.value.getYear.toString
+              }
+            }
+          )
+
+        }
+
+        "the user is starting a draft return and" when {
+
+          "the user is a personal representative" in {
+            val date = DateOfDeath(LocalDate.now())
+            test(
+              sessionWithStartingNewDraftReturn(
+                IncompleteRepresenteeAnswers.empty.copy(dateOfDeath = Some(date)),
+                Left(PersonalRepresentative)
+              )._1,
+              "representee.dateOfDeath.title",
+              routes.RepresenteeController.enterName(),
+              expectReturnToSummaryLink = false,
+              Some(date)
+            )
+
+          }
+
+          "the section is complete" in {
+            val answers = sample[CompleteRepresenteeAnswers]
+            test(
+              sessionWithStartingNewDraftReturn(answers, Left(PersonalRepresentative))._1,
+              "representee.dateOfDeath.title",
+              routes.RepresenteeController.checkYourAnswers(),
+              expectReturnToSummaryLink = false,
+              answers.dateOfDeath
+            )
+          }
+
+        }
+
+        "the user has already started a draft return and" when {
+
+          "the user is a personal representative" in {
+            val date = DateOfDeath(LocalDate.now())
+            test(
+              sessionWithFillingOutReturn(
+                IncompleteRepresenteeAnswers.empty.copy(dateOfDeath = Some(date)),
+                Left(PersonalRepresentative)
+              )._1,
+              "representee.dateOfDeath.title",
+              routes.RepresenteeController.enterName(),
+              expectReturnToSummaryLink = true,
+              Some(date)
+            )
+          }
+
+          "the section is complete" in {
+            val answers = sample[CompleteRepresenteeAnswers]
+            test(
+              sessionWithFillingOutReturn(answers, Left(PersonalRepresentative))._1,
+              "representee.dateOfDeath.title",
+              routes.RepresenteeController.checkYourAnswers(),
+              expectReturnToSummaryLink = true,
+              answers.dateOfDeath
+            )
+          }
+
+        }
+
+      }
+
+    }
+
+    "handling submitted date of death" must {
```

Comment:
are we missing form error scenarios? There's a thing called `DateErrorScenarios` or something that you can reuse to get that covered